### PR TITLE
feat(aot): creating files in a virtual fs.

### PR DESCRIPTION
**Pull request by hansl**
In addition, reading and using AST on main.ts to figure out the entry module, if not specified.

### CODE REVIEW

File: packages/webpack/src/plugin.ts [hunk 1]
```diff
@@ -58,19 +73,81 @@ export class NgcWebpackPlugin {
...
+      .map(node => node as ts.CallExpression)
+      .filter(call => {
+        const access = call.expression as ts.PropertyAccessExpression;
+        return access.kind == ts.SyntaxKind.PropertyAccessExpression
+            && access.name.kind == ts.SyntaxKind.Identifier
+            && (access.name.text == 'bootstrapModule'
+             || access.name.text == 'bootstrapModuleFactory');
+      });
+
+    if (bootstrap.length != 1
```
Comment (filipesilva) [inline packages/webpack/src/plugin.ts]: What about multiple bootstrap calls? Is this something AoT doesn't support?

File: packages/webpack/src/resource_loader.ts [hunk 2]
```diff
@@ -89,24 +89,26 @@ export class WebpackResourceLoader implements ResourceLoader {
...
+      newSource = vmScript.runInContext(vmContext);
+
+      if (typeof newSource == 'string') {
+        return Promise.resolve(newSource);
+      }
+
+      return Promise.reject('The loader "' + fileName + '" didn\'t return a string.');
     } catch (e) {
+      console.log('------');
+      console.log('------');
```
Comment (filipesilva) [inline packages/webpack/src/resource_loader.ts]: Intentional?

File: packages/webpack/src/plugin.ts (hunk 1 above)
Comment (hansl) [inline packages/webpack/src/plugin.ts]: It doesn't support it. Having multiple bootstraps in a single file is just bad. Please note that in this case, the CLI doesn't support it but you can specify which specific module you want to bootstrap with `entryModule` in your webpack config.

File: packages/webpack/src/resource_loader.ts (hunk 2 above)
Comment (hansl) [inline packages/webpack/src/resource_loader.ts]: Oh noes! Removed.

**Pull request merged**